syne_tune/optimizer/schedulers/transfer_learning/rush.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from typing import Dict, List, Optional
+
+from syne_tune.optimizer.schedulers.hyperband import HyperbandScheduler
+from syne_tune.optimizer.schedulers.transfer_learning import TransferLearningTaskEvaluations, TransferLearningMixin
+
+
+class RUSHScheduler(TransferLearningMixin, HyperbandScheduler):
+    def __init__(
+            self,
+            config_space: Dict,
+            transfer_learning_evaluations: Dict[str, TransferLearningTaskEvaluations],
+            metric: str,
+            points_to_evaluate: Optional[List[Dict]] = None,
+            num_hyperparameters_per_task: Optional[int] = 1,
+            **kwargs
+    ) -> None:
+        """
+        A transfer learning variation of Hyperband which uses previously well-performing hyperparameter configurations
+        as an initialization. The best hyperparameter configuration of each individual task provided is evaluated.
+        The one among them which performs best on the current task will serve as a hurdle and is used to prune
+        other candidates. This changes the standard successive halving promotion as follows. As usual, only the top-
+        performing fraction is promoted to the next rung level. However, these candidates need to be at least as good
+        as the hurdle configuration to be promoted. In practice this means that much fewer candidates can be promoted.
+
+        Reference: A resource-efficient method for repeated HPO and NAS.
+        Giovanni Zappella, David Salinas, Cédric Archambeau. AutoML workshop @ ICML 2021.
+
+        :param config_space: configuration space for trial evaluation function.
+        :param metric: objective name to optimize, must be present in transfer learning evaluations.
+        :param transfer_learning_evaluations: dictionary from task name to offline evaluations.
+        :param points_to_evaluate: when points_to_evaluate is not None, the provided configurations are evaluated first
+        in addition to top performing configurations from other tasks and also serve to preemptively prune
+        underperforming configurations
+        :param num_hyperparameters_per_task: the number of top hyperparameter configurations to consider per task.
+        """
+        self._metric_names = [metric]
+        kwargs['type'] = 'rush_stopping'
```

Comment:
Yes, eventually it will have a type that allows for switching between those types. At the moment, there is only one so I've fixed it. I've planned to change as soon as the other one is supported. It's a bit ugly atm, I agree.